rest-server/src/models/job.js

```diff
@@ -94,11 +95,41 @@ class Job {
             next(job, new Error('job not found'));
           } else {
             if (framework.frameworkStatus) {
+              let jobState = "";
+              const frameworkState = framework.frameworkStatus.frameworkState;
+              if (
+                  frameworkState == "FRAMEWORK_WAITING" ||
+                  frameworkState == "APPLICATION_CREATED" ||
+                  frameworkState == "APPLICATION_LAUNCHED" ||
+                  frameworkState == "APPLICATION_WAITING"
+                ) {
+                jobState = "WAITING";
+              } else if (
+                  frameworkState == "APPLICATION_RUNNING" ||
+                  frameworkState == "APPLICATION_RETRIEVING_DIAGNOSTICS" ||
+                  frameworkState == "APPLICATION_COMPLETED"
+                ) {
+                jobState = "RUNNING";
+              } else if (
+                  frameworkState == "FRAMEWORK_COMPLETED"
+                ) {
+                jobState = "COMPLETED";
```

Comment:
>COMPLETED [](start = 28, length = 9)

@github@xiongyf.com and @micw.mm@gmail.com, I assume if jobState == complete, we will further tell if the job is "succeeded" or "failed", right?

Reply:
Currently the state of succeeded or failed is further distinguished in web portal using the appExitStaues field returned by Launcher. Shall we do it here?

Reply:
I don't think we should do it here. but please check with Yuqi if appExitStatues is the legit field to detect the success of a job completion.

---
In reply to: [160622710](https://github.com/Microsoft/pai/pull/210#discussion_r160622710) [](ancestors = 160622710)

Reply:
You can use applicationExitCode to judge succeed/fail when the state = APPLICATION_COMPLETED or FRAMEWORK_COMPLETED

---
In reply to: [160626847](https://github.com/Microsoft/pai/pull/210#discussion_r160626847) [](ancestors = 160626847,160622710)

Reply:
As FRAMEWORK_COMPLETED will arrived finally, so I suggest set jobState to SUCCEEDED or FAILED here depends on applicationExitCode, this will be more user friendly on web portal.

Reply:
Isn't this the current solution?
@hwuu you have to update code in webportal accordingly if changed states here.

Reply:
Thanks guys. I will make the change according to your suggestions.